revscoring/scoring/models/sklearn.py

```diff
@@ -0,0 +1,181 @@
+"""
+Implements the basics of all sklearn based models.
+
+.. autoclass:: revscoring.scoring.models.sklearn.Classifier
+    :members:
+
+.. autoclass:: revscoring.scoring.models.sklearn.ProbabilityClassifier
+    :members:
+"""
+import logging
+import time
+
+from . import model, util
+from ...features import vectorize_values
+from ..statistics import Classification
+
+logger = logging.getLogger(__name__)
+
+
+class Classifier(model.Classifier, model.Model):
+    Estimator = NotImplemented
+    BASE_PARAMS = {}
+    SCORE_SCHEMA = {
+        'title': "Scikit learn-based classifier score",
+        'type': "object",
+        'properties': {
+            'prediction': {
+                'description': "The most likely label predicted by the " +
+                               "estimator",
+                'type': ['string', 'boolean', 'number']
+            }
+        }
+    }
+
+    def __init__(self, features, version=None,
+                 labels=None, label_weights=None, population_rates=None,
+                 scale=False, center=False, statistics=None,
+                 estimator=None, **estimator_params):
+        statistics = statistics or Classification(
+            "prediction", labels=labels, population_rates=population_rates)
+        super().__init__(
+            features, version=version, labels=labels,
+            population_rates=population_rates, scale=scale, center=center,
+            statistics=statistics)
+        self.label_weights = label_weights
+
+        if estimator is None:
+            params = dict(self.BASE_PARAMS)
+            params.update(estimator_params)
+            self.estimator_params = params
+            self.estimator = self.Estimator(**params)
+        else:
+            self.estimator = estimator
+            self.estimator_params = estimator.get_params()
+
+        self.params.update(self.estimator.get_params())
+        self.params.update({'label_weights': label_weights})
+
+    def _clean_copy(self):
+        cls = self.__class__
+        kwargs = dict(self.estimator_params)
+        kwargs.update(self.params)
+        return cls(self.features, version=self.version,
+                   **kwargs)
+
+    def train(self, values_labels, **kwargs):
+        """
+        :Returns:
+            A dictionary with the fields:
+
+            * seconds_elapsed -- Time in seconds spent fitting the model
+        """
+        logger.info("Training {0} with {1} observations"
+                    .format(self.__class__.__name__, len(values_labels)))
+        start = time.time()
+        values, labels = zip(*values_labels)
```

Comment:
Definitely not novel.  I can see that it's unexpected though.  This is really common among python stuff.  See the docs https://docs.python.org/3/library/functions.html#zip

> zip() in conjunction with the * operator can be used to unzip a list

And it even includes a demo!